### PR TITLE
Provinces and Pops loading

### PR DESCRIPTION
**Pull request by IhateTrains**

### CODE REVIEW

File: ImperatorToCK3/Source/Imperator/Provinces/Pops.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	registerKeys();
+	parseStream(theStream);
+	clearRegisteredKeywords();
+}
+
+void ImperatorWorld::Pops::registerKeys()
+{
+	registerRegex("\\d+", [this](const std::string& thePopID, std::istream& theStream) {
+		const auto popStr = commonItems::singleItem(thePopID, theStream);
+		if (popStr.find('{') != std::string::npos)
```
Comment (Zemurin) [inline ImperatorToCK3/Source/Imperator/Provinces/Pops.cpp:18]: Do you expect you might have [popid] = none again? Jesus, PDX.

File: ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvincesTests.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	const ImperatorWorld::Provinces provinces(input);
+	const auto& provinceItr = provinces.getProvinces().find(42);
+	const auto& provinceItr2 = provinces.getProvinces().find(43);
+
+	ASSERT_EQ(42, provinceItr->first);
+	ASSERT_EQ(42, provinceItr->second->getID());
+	ASSERT_EQ(43, provinceItr2->first);
+	ASSERT_EQ(43, provinceItr2->second->getID());
+}
```
Comment (Zemurin) [inline ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvincesTests.cpp:36]: Missing tests for linkage. 
Show what happens when linkage breaks. CK2 has some relevant test templates for this, as well as for checking the error messages that are output.

File: ImperatorToCK3/Source/Imperator/Provinces/Pops.cpp (hunk 1 above)
Comment (IhateTrains) [reply ImperatorToCK3/Source/Imperator/Provinces/Pops.cpp:18]: I'm surprised you're still surprised 😃:

> 51272={
> 	type="citizen"
> 	culture="pasargadi"
> 	religion="zoroaster"
> }
> 51273={
> 	type="citizen"
> 	culture="pasargadi"
> 	religion="zoroaster"
> }
> 51274=none

File: ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvincesTests.cpp (hunk 2 above)
Comment (IhateTrains) [reply ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvincesTests.cpp]: Done, can you check them? Especially the dots vs arrows.

Comment (Zemurin) [reply ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvincesTests.cpp]: looks solid. All green here, might want to wait for @Idhrendur in case I missed something.

**Comment by Idhrendur** [comment]
I'll happily review, but it'll have to be after work. If you don't want to wait the nine hours, feel free to go ahead.

File: ImperatorToCk3Tests/ImperatorWorldTests/Provinces/PopsTests.cpp [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	input << "{\n";
+	input << "}";
+
+	ImperatorWorld::Pops pops;
+	pops.loadPops(input);
+
+	ASSERT_TRUE(pops.getPops().empty());
+}
+
+TEST(ImperatorWorld_PopsTests, popsCanBeLoaded)
```
Comment (Idhrendur) [inline ImperatorToCk3Tests/ImperatorWorldTests/Provinces/PopsTests.cpp:20]: You should have your input also have a representation of the none form. If that's meant to be excluded, instead have a test demonstrating it gets excluded.

File: ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvinceTests.cpp [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	input << "=\n";
+	input << "{\n";
+	input << "}";
+
+	const ImperatorWorld::Province theProvince(input, 42);
+
+	ASSERT_TRUE(theProvince.getName().empty());
+}
+
+TEST(ImperatorWorld_ProvinceTests, ownerCanBeSet)
```
Comment (Idhrendur) [inline ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvinceTests.cpp:95]: Add test for default owner

File: ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvincesTests.cpp [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+	input << "=\n";
+	input << "{\n";
+	input << "}";
+
+	const ImperatorWorld::Provinces provinces(input);
+
+	ASSERT_TRUE(provinces.getProvinces().empty());
+}
+
+TEST(ImperatorWorld_ProvincesTests, provincesCanBeLoaded)
```
Comment (Idhrendur) [inline ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvincesTests.cpp:21]: Also add test for the none form here.

Comment (IhateTrains) [reply ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvincesTests.cpp:21]: There was no case of id=none in Provinces, did you mean Families?

File: ImperatorToCk3Tests/ImperatorWorldTests/Provinces/PopsTests.cpp (hunk 3 above)
Comment (IhateTrains) [reply ImperatorToCk3Tests/ImperatorWorldTests/Provinces/PopsTests.cpp:20]: Added ✔️

File: ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvinceTests.cpp (hunk 4 above)
Comment (IhateTrains) [reply ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvinceTests.cpp:95]: Added ✔️

File: ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvincesTests.cpp (hunk 5 above)
Comment (IhateTrains) [reply ImperatorToCk3Tests/ImperatorWorldTests/Provinces/ProvincesTests.cpp:21]: Added (for Families) ✔️
